packages/api/src/controllers/user.ts

```diff
@@ -636,9 +653,105 @@ app.post("/token", validatePost("user"), async (req, res) => {
     });
   }
   res.status(201);
-  res.json({ id: user.id, email: user.email, token: token });
+  res.json({ id: user.id, email: user.email, token, refreshToken });
 });
 
+app.post(
+  "/token/refresh",
+  validatePost("refresh-token-payload"),
+  async (req, res) => {
+    const now = Date.now();
+    const { refreshToken } = req.body as RefreshTokenPayload;
+    const refreshTokenObj = await db.jwtRefreshToken.get(refreshToken);
+    if (
+      !refreshTokenObj ||
+      refreshTokenObj.expiresAt < now ||
+      refreshTokenObj.revoked
+    ) {
+      console.log(
+        `Refresh attempt with invalid refresh token=${refreshToken} expiresAt=${refreshTokenObj?.expiresAt} revoked=${refreshTokenObj?.revoked}`
+      );
+      return res.status(401).json({ errors: ["invalid refresh token"] });
+    }
+
+    if (now - refreshTokenObj.lastSeen < req.config.jwtAccessTokenTtl / 2) {
```

Comment:
What's the use of this line? Why do we revoke token? Do we maybe have a test scenario for this?

Reply:
This is the only use for the `lastSeen` field which I explained here: https://github.com/livepeer/studio/pull/2004#discussion_r1440700535

The logic here is that a single refresh token should not be used to generate multiple access tokens. We could limit this even further to only allow 1 to be generated every `jwtAccessTokenTtl`, but I added a `/2` in there just to give it a bit of leeway and avoid random clock skew errors (so one could generate 1 access tokens every 12h, lasting 24h each).

I'll add a comment here and maybe restructure the expression to be clearer.

Reply:
Maybe stupid question, I'm not a frontend person, but when you have multiple tabs opened, then don't each need a separate JWT token? What happens if one tab refreshes the token, but we don't allow multiple refreshes?